src/master/develop/connection.md

```diff
@@ -51,8 +51,35 @@ immuClient.login("immudb", "immudb");
 :::
 
 ::: tab Python
-This feature is not yet supported or not documented.
-Do you want to make a feature request or help out? Open an issue on [Python sdk github project](https://github.com/codenotary/immudb-py/issues/new)
+
+```python
+from grpc import RpcError
+from immudb import ImmudbClient
+
+URL = "localhost:3322"  # immudb running on your machine
+LOGIN = "immudb"        # Default username
+PASSWORD = "immudb"     # Default password
+DB = b"defaultdb"       # Default database name (must be in bytes)
+
+
+def main():
+    client = ImmudbClient(URL)
+    # database parameter is optional
+    client.login(LOGIN, PASSWORD, database=DB)
+    client.logout()
+
+    # Bad login
+    try:
+        client.login("verybadlogin", "verybadpassword")
+    except RpcError as exception:
```

Comment:
An idea for future improvement: We should detect what error it is - like here that credentials are incorrect. There should be data sent along the error code from the backend in the details - it could be used to convert to appropriate specific exception.

Reply:
Definitely it should cast to some normal Exception

Unfortunately from my investigation on it - this exception have mainly 3 "normal" fields now

debug_error_string ```{"created":"@1649922703.326925387","description":"Error received from peer ipv6:[::1]:3322","file":"src/core/lib/surface/call.cc","file_line":903,"grpc_message":"invalid user name or password","grpc_status":2}```

details ```invalid user name or password```

code ```StatusCode.UNKNOWN``` - it matches internal codes from GRPC, and not the code provided by server (08004)


I looked depper into it and i see that the proper status code is in some unparsed trailing_metadata field
```(_Metadatum(key='grpc-status-details-bin', value=b'\x08\x02\x12\x1dinvalid user name or password\x1aE\n+type.googleapis.com/immudb.schema.ErrorInfo\x12\x16\n\x0508004\x12\rkey not found\x1a-\n+type.googleapis.com/immudb.schema.RetryInfo'),)```

```.....schema.ErrorInfo\x12\x16\n\x05------->08004<-------\x12\rkey not found\x1a-\n+type.google......```


I will write an issue also

Reply:
https://github.com/codenotary/immudb-py/issues/40

Reply:
Last year, I suggested immudb should report error codes. Error codes are used by many solutions in different areas, are language agnostic and are extensible. I remember Duncan worked on that and created a proposal. Can we continue where he stopped?